plugins/analytics-module-ga/src/apis/implementations/AnalyticsApi/GoogleAnalytics.ts

```diff
@@ -0,0 +1,160 @@
+/*
+ * Copyright 2021 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import ReactGA from 'react-ga';
+import {
+  AnalyticsApi,
+  AnalyticsContextValue,
+  AnalyticsEventAttributes,
+  AnalyticsEvent,
+} from '@backstage/core-plugin-api';
+import { Config } from '@backstage/config';
+
+type CDM = {
+  type: 'dimension' | 'metric';
+  index: number;
+  source: 'context' | 'attributes';
+  key: string;
+};
+
+/**
+ * Type guard for emptiness check in array filter.
+ */
+function notEmpty<T>(value: T | undefined): value is T {
+  return value !== undefined;
+}
+
+/**
+ * Google Analytics API provider for the Backstage Analytics API.
+ */
+export class GoogleAnalytics implements AnalyticsApi {
+  private readonly cdmConfig: CDM[];
+
+  /**
+   * Instantiate the implementation and initialize ReactGA.
+   */
+  private constructor({
+    cdmConfig,
+    trackingId,
+    testMode,
+    debug,
+  }: {
+    cdmConfig: CDM[];
+    trackingId: string;
+    testMode: boolean;
+    debug: boolean;
```

Comment:
I think the `fromConfig()` always ends up passing a boolean, regardless of what you have configured, and the constructor itself is private otherwise.  Kept as-is.  Thanks!